lms/djangoapps/courseware/tests/test_entrance_exam.py

```diff
@@ -124,13 +126,39 @@ def setUp(self):
             user,
             self.entrance_exam
         )
-        self.entrance_exam.is_entrance_exam = True
-        self.entrance_exam.in_entrance_exam = True
         self.course.entrance_exam_enabled = True
         self.course.entrance_exam_minimum_score_pct = 0.50
         self.course.entrance_exam_id = unicode(self.entrance_exam.scope_ids.usage_id)
         modulestore().update_item(self.course, user.id)  # pylint: disable=no-member
 
+    def test_overriding_chapter_with_required_content_module(self):
+        """
+        Unit Test: if entrance exam is required then show its content e.g. chapter, sub-section.
+        """
+        field_data_cache = FieldDataCache.cache_for_descriptor_descendents(
+            self.course.id,
+            self.request.user,
+            self.course,
+            depth=2
+        )
+        # pylint: disable=protected-access
+        course_module = get_module(
+            self.request.user,
+            self.request,
+            self.course.scope_ids.usage_id,
+            field_data_cache,
+        )._xmodule
+
+        chapter, section = override_with_required_content(
+            course_module=course_module,
+            course=self.course,
+            user=self.request.user,
+            active_chapter=self.chapter.url_name,
+            active_section=self.chapter_subsection.url_name
+        )
+        self.assertEqual(chapter, self.entrance_exam.url_name)
+        self.assertEqual(section, self.exam_1.url_name)
+
```

Comment:
Can we add some tests where we use the views.py, e.g. make a web request and make sure we see the gated (entrance exam) content, by inspecting the HTML content?